src/main/java/org/apache/hadoop/mapred/ResourcePolicy.java

```diff
@@ -383,16 +383,22 @@ public void resourceOffers(SchedulerDriver schedulerDriver,
         }
 
         String uri = scheduler.conf.get("mapred.mesos.executor.uri");
-        if (uri == null) {
+        String directory = scheduler.conf.get("mapred.mesos.executor.directory");
+        boolean isUriSet = uri != null && !uri.equals("");
+        boolean isDirectorySet = directory != null && !directory.equals("");
+
+        if (!isUriSet && !isDirectorySet) {
           throw new RuntimeException(
               "Expecting configuration property 'mapred.mesos.executor'");
-        }
-
-        String directory = scheduler.conf.get("mapred.mesos.executor.directory");
-        if (directory == null || directory.equals("")) {
+        } else if (isUriSet && isDirectorySet) {
+          throw new RuntimeException(
+              "Conflicting properties 'mapred.mesos.executor.uri' and 'mapred.mesos.executor.directory', only one can be set");
```

Comment:
I think so, yeah... It might be that users want to specify a sub directory of the URI that's downloaded and extracted to use instead of just `bin/hadoop` at the root, but i'm not entirely sure.